### DISCUSSION TITLE
Crazyradio 2.0 not Detected in CFCLIENT python app

### DISCUSSION THREAD

**Discussion by amitx64** [bitcraze/discussions#1026] [Q&A] [answered]
@knmcguire and @ataffanel, I'm encountering an issue with my `crazyradio`. I've uploaded both `crazyradio2-bootloader` and `crazyradio2-firmware` onto my newly designed `crazyradio` following the Bitcraze schematics. However, upon plugging the `crazyradio dongle` into my PC, it's detected as `Bitcraze AB Crazyradio 2.0`. I've noticed in some YouTube videos that the detection shows as `crazyradio USB Dongle`. Additionally, when I attempt to run the Crazyflie client application, it fails to detect my device. Consequently, I'm unable to connect my crazyradio dongle with your Python app, and I'm receiving the error message: "CrazyRadio Dongle not found." Any assistance you can provide would be greatly appreciated.
 
 I just connected the 'Crazyradio2.0' to the PC and attempted to detect the 'Crazyflie2.0' using the 'cfclient' app. Should I power on the 'Crazyflie' drone? If so, does it mean that the 'cfclient' will detect the 'Crazyflie' drone and establish a connection via the Crazyradio? Please correct me if I'm wrong.
 
 ERROR:
 ```
 INFO:__main__:Disabling STL printouts

INFO:cfclient.utils.input.inputreaders:Input readers: ['linuxjsdev', 'pysdl2']

INFO:cfclient.utils.input.inputreaders:Successfully initialized [linuxjsdev]

INFO:cfclient.utils.input.inputreaders:Could not initialize [pysdl2]: No SDL2 support on Linux

INFO:cfclient.utils.input.inputinterfaces:Found interfaces: ['leapmotion', 'wiimote', 'zmqpull']

INFO:cfclient.utils.input.inputinterfaces:Could not initialize [leapmotion]: Leap Motion library probably not installed (No module named 'leapsdk')

INFO:cfclient.utils.input.inputinterfaces:Could not initialize [wiimote]: Missing cwiid (wiimote) driver No module named 'cwiid'

INFO:cfclient.utils.config:Dist config read from /home/bitcraze/projects/crazyflie-clients-python/src/cfclient/configs/config.json

INFO:cfclient.utils.config:Config file read from [/home/bitcraze/.config/cfclient/config.json]

INFO:cfclient.utils.input.inputinterfaces:Could not initialize [zmqpull]: ZMQ input disabled in config file

INFO:cfclient.utils.zmq_led_driver:ZMQ led disabled in config file

INFO:cfclient.utils.zmq_param:ZMQ param disabled in config file

INFO:__main__:If startup fails because of "xcb", install dependency with `sudo apt install libxcb-xinerama0`.

INFO:cfclient.utils.input:Using device blacklist [(VirtualBox|VMware)]

Cannot find a Crazyradio Dongle

INFO:cflib.drivers.cfusb:Looking for devices....

INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas

INFO:cfclient.ui.tab_toolbox:No config found for open_toolboxes

INFO:cfclient.ui.tab_toolbox:No config found for open_toolboxes

WARNING:cfclient.ui.main:Cannot find a Crazyradio Dongle

Cannot find a Crazyradio Dongle

INFO:cflib.drivers.cfusb:Looking for devices....

INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.

WARNING:cfclient.ui.main:Cannot find a Crazyradio Dongle

Cannot find a Crazyradio Dongle

INFO:cflib.drivers.cfusb:Looking for devices....

INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.

```

**Comment by ataffanel** [comment, marked as answer]
Hi,

If you built the firmware by yourself make sure you are building the compatibility firmware. The Crazyradio 2 firmware release page only has the compatibility firmware available since this is the only one currently supported by the Crazyflie libs and clients.

To make sure you have the compatibility firmware, Crazyradio should appear on USB as device `1915:7777`:
```
❯ lsusb -d 1915:7777
Bus 009 Device 018: ID 1915:7777 Nordic Semiconductor ASA Bitcraze Crazyradio (PA) dongle
```

If it appears instead as `35F0:AD20`, you are running the new usb protocol which is not currently compatible with any lib or client.

**Reply by amitx64** [reply]
Thank you, @ataffanel, for sharing valuable information. Could you please clarify whether I need to update both the Bootloader and Firmware of the Crazyradio 2.0, or just the Firmware? Additionally, if possible, could you provide the binary files for both the Bootloader and Firmware along with the relevant git branch details? This would help me avoid any mistakes in the future. Thanks again.

**Reply by amitx64** [reply]
@ataffanel, following your suggestion, I uploaded the Firmware from the release. Now, when I execute `lsusb -d 1915:7777`, it shows the correct result.
```
ID 1915:7777 Nordic Semiconductor ASA Bitcraze Crazyradio (PA) dongle
```
However, when I run cfclient, the following output is displayed:

```
INFO:__main__:Disabling STL printouts
INFO:cfclient.utils.input.inputreaders:Input readers: ['linuxjsdev', 'pysdl2']
INFO:cfclient.utils.input.inputreaders:Successfully initialized [linuxjsdev]
INFO:cfclient.utils.input.inputreaders:Could not initialize [pysdl2]: No SDL2 support on Linux
INFO:cfclient.utils.input.inputinterfaces:Found interfaces: ['leapmotion', 'wiimote', 'zmqpull']
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [leapmotion]: Leap Motion library probably not installed (No module named 'leapsdk')
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [wiimote]: Missing cwiid (wiimote) driver No module named 'cwiid'
INFO:cfclient.utils.config:Dist config read from /home/bitcraze/projects/crazyflie-clients-python/src/cfclient/configs/config.json
INFO:cfclient.utils.config:Config file read from [/home/bitcraze/.config/cfclient/config.json]
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [zmqpull]: ZMQ input disabled in the config file
INFO:cfclient.utils.zmq_led_driver:ZMQ led disabled in the config file
INFO:cfclient.utils.zmq_param:ZMQ param disabled in the config file
INFO:__main__:If startup fails because of "xcb," install the dependency with `sudo apt install libxcb-xinerama0`.
INFO:cfclient.utils.input:Using device blacklist [(VirtualBox|VMware)]
INFO:cfclient.ui.tab_toolbox:No config found for toolbox_areas
INFO:cfclient.ui.tab_toolbox:No config found for open_toolboxes
INFO:cflib.crtp.radiodriver:v3.2 dongle with serial N/A found
INFO:cflib.drivers.cfusb:Looking for devices....
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.

```

Could you please confirm whether this output indicates that everything is correct now? Thank you.

**Reply by knmcguire** [reply]
hi! Let me step in here but this line indicates that it is able to find the crazyradio now!

```
INFO:cflib.crtp.radiodriver:v3.2 dongle with serial N/A found
```

Can you connect to the crazyflie now ?

**Reply by amitx64** [reply]
@knmcguire Actually, I designed the `Crazyflie 2.1` and `Crazyflie Bolt` according to the `Bitcraze`  schematics. Currently, I am working on uploading the respective `Bootloader` and `Firmware` to both NRF51 and STM32F405. Following this, I aim to establish a connection between the `Crazyflie` and the `cfclient` Python app using `Crazyradio`. However, I am facing an issue where the NRF51 on the `Crazyflie` board I designed is not connecting with my `Stlink`. I am actively investigating the PCB to identify and resolve the issue. I will update you once I have more information.

**Reply by amitx64** [reply]
@knmcguire My goal is to enable indoor drone flight with animations, similar to what `Bitcraze` demonstrated in this video:
```
https://www.youtube.com/watch?v=w8jAHYlcj7k
```

**Reply by knmcguire** [reply]
hmm oke. that is going to be more challenging for us to help if this is custom  hardware and unfortunately out of the scope of this forum. In terms of the NRF51 not connecting you should probably contract nordic semiconductors directly.

**Reply by amitx64** [reply]
@knmcguire Yes, I am actively working on it. I have designed my circuit based on the schematics provided by `Bitcraze`.

**Reply by amitx64** [reply]
Hi, @knmcguire and @ataffanel. I've designed the flight controller `Crazyflie 2.1` based on the Bitcraze schematic. After uploading `crazyflie2-nrf-bootloader`, `crazyflie2-nrf-firmware`, and `crazyflie2-stm-bootloader`, I attempted to connect the `Crazyradio` to `Crazyflie 2.1` using CFClient. However, the CFClient did not detect `Crazyflie 2.1`. Here, is the result: 

![Screenshot from 2023-11-15 17-49-19](https://github.com/bitcraze/discussions/assets/79006087/30ea5161-dc17-48e7-a075-dbb01ec1f2fe)

**Reply by amitx64** [reply]
Hey @knmcguire and @ataffanel, could you please confirm if I'm following the correct steps to configure the `Crazyflie 2.1` Flight controller? Here's what I've done so far:

1. Uploaded the `nrf51` bootloader using the `openocd` command. (However, I don't know the starting address.)
2. Uploaded the `nrf51` firmware using the `openocd` command. (Again, I don't know the starting address.)
3. Uploaded `crazyflie2-stm-bootloader` on the STM32F405 using the `openocd` command.
4. Finally, I attempted to connect my `Crazyflie 2.1` Flight controller with CFClient, but I wasn't successful.

Can you help me figure out what might be causing the issue?

**Reply by knmcguire** [reply]
So, we can't really confirm anything as this might be hardware related and we don't have an idea of what you have done in the process. As I mentioned before, this is outside of the scope of our discussion platform.

In the meantime, I'd like to refer you to the answer that Arnaud has compiled for another user who has a similar question: https://github.com/orgs/bitcraze/discussions/1037

**Reply by amitx64** [reply]
@knmcguire, I am conducting this for research purposes. This involves designing a circuit based on the Bitcraze schematic and ensuring functionality through the utilization of the `bootloader`, `firmware`, and software provided by Bitcraze. The ultimate goal is to create a small drone light swarming system, similar to what Bitcraze has achieved.
